packages/react-components/react-conformance-griffel/src/overridesWin.ts

```diff
@@ -73,21 +72,56 @@ export const overridesWin: BaseConformanceTest = testInfo => {
     // https://github.com/facebook/jest/issues/8987#issuecomment-584898030
     /* eslint-disable @fluentui/no-global-react */
     const React = await import('react');
-    const ReactDOM = await import('react-dom');
 
     const className = 'make-styles-classname';
     const Component = await getReactComponent(testInfo.componentPath, testInfo);
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const element = React.createElement(Component, { ...testInfo.requiredProps, className } as any);
 
-    ReactDOM.render(element, container);
+    const { unmount } = await render(element, container as HTMLElement);
 
     expect(mergeClasses.mock.calls.length).toBeGreaterThanOrEqual(1);
     expect(mergeClasses.mock.calls).toContainClassNameLastInCalls(className);
     expect(mergeClasses.mock.calls).toHaveMergeClassesCalledTimesWithClassName(
       className,
       testOptions?.[OVERRIDES_WIN_TEST_NAME]?.callCount || 1,
     );
+
+    unmount();
   });
 };
+
+/**
+ * Utility to render React elements that works with both React 17 and React 18
+ */
+async function render(element: React.ReactElement, container: HTMLElement) {
```

Comment:
we cannot use `@testing-library/react` as this conformance test has its own perils that can be read within the  comments related to mocking etc.

TL;DR:

RTL already declares its onw after/before Each hooks, and we cannot do these hooks within `it` specification, thus the solution is manual logic for booting used react version within the test